alerts/alerts_test.go

```diff
@@ -332,122 +321,62 @@ func TestMessageFilter(t *testing.T) {
 		Filter:               []string{"", "*(test)^"},
 		ServiceAlertPriority: 4,
 		SystemAlertPriority:  3,
 	}},
-		&serviceProvider, &cursorStorage)
+		&instanceProvider, &cursorStorage)
 	if err != nil {
 		t.Fatalf("Can't create alerts: %s", err)
 	}
 	defer alertsHandler.Close()
 }
 
-func TestOtheralerts(t *testing.T) {
+func TestAlertQueueLimit(t *testing.T) {
+	testJournal := testSystemdJournal{}
+	alerts.CurrentJournal = &testJournal
+
 	alertsHandler, err := alerts.New(&config.Config{
 		Alerts: config.Alerts{
 			ServiceAlertPriority: 4,
 			SystemAlertPriority:  3,
 		},
 	},
-		&serviceProvider, &cursorStorage)
+		&instanceProvider, &cursorStorage)
 	if err != nil {
 		t.Fatalf("Can't create alerts: %s", err)
 	}
 	defer alertsHandler.Close()
 
-	time := time.Now()
-	alertsHandler.SendResourceAlert("test", "testResource", time, 42)
-
-	resourceAlert := pb.Alert{
-		Timestamp: timestamppb.New(time),
-		Tag:       alerts.AlertTagResource,
-		Source:    "test",
-		Payload: &pb.Alert_ResourceAlert{ResourceAlert: &pb.ResourceAlert{
-			Parameter: "testResource",
-			Value:     42,
-		}},
-	}
-
-	if err = waitResult(alertsHandler.GetAlertsChannel(), 5, func(alert *pb.Alert) (success bool, err error) {
-		if !proto.Equal(alert, &resourceAlert) {
-			return false, aoserrors.New("received resource alert != send alert")
-		}
-
-		return true, nil
-	}); err != nil {
-		t.Errorf("Result failed: %s", err)
-	}
-
-	deviceErrors := make(map[string][]error)
-	deviceErrors["dev1"] = []error{aoserrors.New("some error")}
-
-	alertsHandler.SendValidateResourceAlert("test", deviceErrors)
-
-	convertedErrors := make([]*pb.ResourceValidateErrors, 0)
-
-	for name, reason := range deviceErrors {
-		var messages []string
-
-		for _, item := range reason {
-			messages = append(messages, item.Error())
-		}
-
-		resourceError := pb.ResourceValidateErrors{
-			Name:     name,
-			ErrorMsg: messages,
-		}
-
-		convertedErrors = append(convertedErrors, &resourceError)
+	for i := 1; i < 55; i++ {
+		alertsHandler.SendAlert(cloudprotocol.AlertItem{
+			Tag:     cloudprotocol.AlertTagSystemError,
+			Payload: cloudprotocol.SystemAlert{Message: "some error"},
+		})
 	}
 
-	validationAlert := pb.Alert{
-		Timestamp: timestamppb.New(time),
-		Tag:       alerts.AlertTagAosCore,
-		Source:    "test",
-		Payload: &pb.Alert_ResourceValidateAlert{
-			ResourceValidateAlert: &pb.ResourceValidateAlert{
-				Type:   alerts.AlertDeviceErrors,
-				Errors: convertedErrors,
-			},
-		},
-	}
-
-	if err = waitResult(alertsHandler.GetAlertsChannel(), 5, func(alert *pb.Alert) (success bool, err error) {
-		alert.Timestamp = validationAlert.Timestamp
-		if !proto.Equal(alert, &validationAlert) {
-			return false, aoserrors.New("received validation alert != send alert")
-		}
-
-		return true, nil
-	}); err != nil {
-		t.Errorf("Result failed: %s", err)
+	if len(alertsHandler.GetAlertsChannel()) != 50 {
+		t.Error("incorrect channel size ", len(alertsHandler.GetAlertsChannel()))
 	}
 }
 
```

Comment:
Would be nice to check if API in `setupJournal` is set correctly. For example, PRIORITY is set according to configured version, etc.

Reply:
done